docs/MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,212 @@
+# Migrating to webpack-blocks v1.0
+
+## Table of Contents
+
+<!-- To update run: npx markdown-toc --maxdepth 2 -i docs/MIGRATION-GUIDE.md -->
+
+<!-- toc -->
+
+- [Migrating configuration](#migrating-configuration)
+- [Additional features](#additional-features)
+- [Migrating blocks](#migrating-blocks)
+
+<!-- tocstop -->
+
+
+## Migrating configuration
+
+### `match()`
+
+You can now use `match()` to specify on which files to apply certain loaders. Works with every block that adds a loader, like `babel`, `css`, `elm`, `postcss`, `sass`, ...
+
+The blocks still work without `match()`. **However, blocks dropped the `fileType`, `exclude` & `include` from their options. Use `match()` for that.**
+
+```js
+const { createConfig, css, file, match, postcss, url } = require('webpack-blocks')
+const path = require('path')
+
+module.exports = createConfig([
+  babel(),            // matches *.js, *.jsx outside node_modules/ by default
+  match('*.css', { exclude: path.resolve('node_modules') }, [
+    css(),
+    postcss()
+  ]),
+  match(['*.gif', '*.jpg', '*.jpeg', '*.png', '*.svg', '*.webp'], [
+    file()
+  ])
+])
+```
+
+### `@webpack-blocks/assets` package
+
+The `assets` package contains the `css`, `css.modules`, `file` and `url` blocks which were contained in `@webpack-blocks/webpack` in v0.4. Use these blocks to configure a `css-loader`, `file-loader` or `url-loader`.
+
+See the package's README for further details.
+
+### `webpack-blocks` package
+
+The new `webpack-blocks` package can be installed using `npm install --save-dev webpack-blocks`. It is a convenience package, wrapping and exporting most of frequently used blocks, so you don't have to manage all those small package dependencies by yourself. Contains:
+
+* assets
+* babel
+* dev-server
+* extract-text
+* postcss
+* sass
+* typescript
+* uglify
+* webpack
+
+It also exports all the `@webpack-blocks/webpack` utility functions. See the package's README for further details.
+
+### No more webpack v1
+
+Support for webpack 1.x has been dropped. If you have not upgraded yet we recommend doing so soon.
+
+### webpack as a peer dependency
+
+Webpack is now a peer dependency of `@webpack-blocks/webpack`. Just make sure to explicitly `npm install --save-dev webpack`.
+
+### babel6 is now babel
+
+The `@webpack-blocks/babel6` package has been renamed to `@webpack-blocks/babel`. You can now choose which version of babel to use, since `babel-core` is now a peer dependency (see below).
+
+### babel-core as a peer dependency
+
+`babel-core` is now a peer dependency of the `babel` block, so you can control which version of `babel-core` to use. Make sure to `npm install --save-dev babel-core` in your application.
+
+### `webpack` no longer exported
+
+The `webpack` instance is no longer exported by `@webpack-blocks/webpack`. Use `const webpack = require('webpack')` instead which is more explicit.
```

Comment:
I think the term “instance” is used incorrectly here because it’s a reexport of the whole `webpack` module.